Add transition between screens

NaughtyChas opened the pull request:
I practice my coding skills on LeetCode every day and I can write shitcode now. Bro is feeling a bit too wired.

Maybe we have to put this as a draft for a long time... Indeed it is a quality-of-life feature for streamers but currently not that easy to get this implemented.

![c64eade6c56be67e1ce39f3e41bde061](https://github.com/user-attachments/assets/e6d4bae2-0145-48f5-96a6-4c98e3508ab8)

NaughtyChas committed:
Scene transition?

there is a big black fking box on the screen and I hate it. Navigate to gameplay to address this.

NaughtyChas committed:
Simple code formatting

NaughtyChas committed:
Trying to figure out how tests works

CloneWith commented:
The *Code Quality* CI is equivalent to the `InspectCode.ps1` script in the project root directory. Since it's very slow, using IDEs like VS Code or Rider should get similar results with faster speed (but not guaranteed). Let alone unit tests through the whole project... Then you'll know why most PRs in the upstream project have ❌s... Visually annoying, but needed for maintaining code quality and finding potential bugs (that's when I learned to write tests along with normal implementations).

NaughtyChas committed:
Merge branch 'derived/new' into feature/transition-merge

NaughtyChas committed:
Merge from master

NaughtyChas committed:
Refactor TournamentSceneManager.cs

I'm surprised that the animation of the chatbox background was defined here. Refactored the animation to a method so it can be easily used in delayed and non-delayed transitioning.

NaughtyChas committed:
Add transitions to BackgroundTypes

we might need a better file name after this...

NaughtyChas committed:
Filetype check for Transitions

NaughtyChas committed:
Restrict transition playback in StreamContainer
